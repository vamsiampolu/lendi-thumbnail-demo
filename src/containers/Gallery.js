import React from 'react'
import Loading from '../components/Loading'
import Thumbnail from '../components/Thumbnail'
import { Card } from 'semantic-ui-react'
import { connect } from 'react-redux'

export function SimpleGallery(props) {
  if(props.isFetching) {
    return (<Loading />)
  } else if(props.albums) {
    const thumbnails = props.albums.map(item => (<Thumbnail key={item.albumId + '#' + item.id} {...item} />))
    return (<Card.Group>
      {thumbnails}
    </Card.Group>)
  } else if(props.error) {
    return (<div>An error occured</div>)
  }
}

const mapStateToProps = (state) => ({
    albums: state.albums,
    isFetching: state.isFetching,
    error: state.error
})

const Gallery = connect(mapStateToProps)(SimpleGallery)

const mapStateToAlbumProps = (state, ownProps) => {
  if(state.albums && state.albums.length) {
    const albums = state.albums.filter(item => item.albumId == ownProps.params.albumId)
    return {
      isFetching: state.isFetching,
      albums,
      error: state.error
    }
  } else {
      return {
        isFetching: state.isFetching,
        error: state.error
      }
  }
}

export const Album = connect(mapStateToAlbumProps)(SimpleGallery)

export default Gallery
