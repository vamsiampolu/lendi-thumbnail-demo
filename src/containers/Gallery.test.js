import React from 'react'
import { SimpleGallery } from './Gallery'
import renderer from 'react-test-renderer'

describe('Gallery', () => {
  it('renders correctly whe Loading', () => {
    const propsLoading = {
      isFetching: true
    }

    const component = renderer.create(<SimpleGallery {...propsLoading} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with Thumbnails', () => {
    const albums = [
    {
      "albumId": 1,
      "id": 1,
      "title": "accusamus beatae ad facilis cum similique qui sunt",
      "url": "http://placehold.it/600/92c952",
      "thumbnailUrl": "http://placehold.it/150/30ac17"
    },
    {
      "albumId": 1,
      "id": 2,
      "title": "reprehenderit est deserunt velit ipsam",
      "url": "http://placehold.it/600/771796",
      "thumbnailUrl": "http://placehold.it/150/dff9f6"
    },
    {
      "albumId": 1,
      "id": 3,
      "title": "officia porro iure quia iusto qui ipsa ut modi",
      "url": "http://placehold.it/600/24f355",
      "thumbnailUrl": "http://placehold.it/150/1941e9"
    }
    ]
    const propsImages = {
      albums,
      isFetching: false
    }
    const component = renderer.create(<SimpleGallery {...propsImages} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should display error correctly', () => {
    const props = {
      error: new Error('something aweful happened')
    }

    const component = renderer.create(<SimpleGallery {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
