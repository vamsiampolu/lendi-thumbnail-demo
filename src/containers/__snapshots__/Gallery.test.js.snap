exports[`Gallery renders correctly whe Loading 1`] = `null`;

exports[`Gallery renders correctly with Thumbnails 1`] = `
<div
  className="ui cards">
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}>
    <img
      alt={undefined}
      className="ui image"
      height={undefined}
      src="http://placehold.it/150/30ac17"
      width={undefined} />
    <div
      className="content">
      <div
        className="header">
        accusamus beatae ad facilis cum similique qui sunt
      </div>
      <div
        className="meta">
        Album 
        1
      </div>
    </div>
    <div
      className="extra content">
      <button
        className="ui green basic button"
        onClick={[Function]}
        tabIndex={undefined}>
        <a
          onClick={[Function]}
          style={Object {}}>
           View Image 
        </a>
      </button>
    </div>
  </div>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}>
    <img
      alt={undefined}
      className="ui image"
      height={undefined}
      src="http://placehold.it/150/dff9f6"
      width={undefined} />
    <div
      className="content">
      <div
        className="header">
        reprehenderit est deserunt velit ipsam
      </div>
      <div
        className="meta">
        Album 
        1
      </div>
    </div>
    <div
      className="extra content">
      <button
        className="ui green basic button"
        onClick={[Function]}
        tabIndex={undefined}>
        <a
          onClick={[Function]}
          style={Object {}}>
           View Image 
        </a>
      </button>
    </div>
  </div>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}>
    <img
      alt={undefined}
      className="ui image"
      height={undefined}
      src="http://placehold.it/150/1941e9"
      width={undefined} />
    <div
      className="content">
      <div
        className="header">
        officia porro iure quia iusto qui ipsa ut modi
      </div>
      <div
        className="meta">
        Album 
        1
      </div>
    </div>
    <div
      className="extra content">
      <button
        className="ui green basic button"
        onClick={[Function]}
        tabIndex={undefined}>
        <a
          onClick={[Function]}
          style={Object {}}>
           View Image 
        </a>
      </button>
    </div>
  </div>
</div>
`;

exports[`Gallery should display error correctly 1`] = `
<div>
  An error occured
</div>
`;
